### PR TITLE
Created IntervalOverlapReadFilter

### LINKED ISSUE #4860: Add an IntervalOverlapReadFilter
Currently the best way to select a subset of reads overlapping an interval is to provide a list of -L reader interval arguments. Unfortunately, this requires that the bam be sorted and indexed, which can often be a pain point when trying to perform quick tests, as a sort is often slow and requires copying the bam a second time. To alleviate this, it would be nice to add a ReadFilter level interval argument, for example an `IntervalOverlapReadFilter` or some such which can take similarly formatted arguments to -L. This would allow one to use `PrintReads` to select an interval over unsorted/unindexed bams more easily.

### PR DESCRIPTION
**Pull request by jamesemery**
I took over #5367 but since I don't have access to your fork @magicDGS I have created a second branch with my changes.

Closes #4860

### CODE REVIEW

**Comment by lbergelson** [comment]
This is failing tests.

File: src/main/java/org/broadinstitute/hellbender/engine/filters/IntervalOverlapReadFilter.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+@DocumentedFeature(groupName= HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY, summary = "Filters out reads that don't overlap the specified region. NOTE: This approach to extracting overlapping reads is very slow compared to using PrintReads and -L on an indexed bam file.")
+public final class IntervalOverlapReadFilter extends ReadFilter {
+    private static final long serialVersionUID = 1L;
+
+    // Interval list files such as Picard interval lists are structured and require specialized parsing that
+    // is handled by IntervalUtils, so use suppressFileExpansion to bypass command line parser auto-expansion.
+    @Argument(fullName = ReadFilterArgumentDefinitions.KEEP_INTERVAL_NAME, suppressFileExpansion = true, doc = "One or more genomic intervals to keep", optional = false)
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/engine/filters/IntervalOverlapReadFilter.java]: I don't know if you want to deal correctly with the special `-L UNMAPPED` input.  If you do you'll have to handle it somewhere...

Comment (jamesemery) [reply src/main/java/org/broadinstitute/hellbender/engine/filters/IntervalOverlapReadFilter.java]: I added a test, it turns out the real fix was in our hearts all along!

**Comment by jamesemery** [comment]
@lbergelson Is this good to merge when tests pass?

File: src/main/java/org/broadinstitute/hellbender/engine/filters/IntervalOverlapReadFilter.java (hunk 1 above)
Comment (lbergelson) [reply src/main/java/org/broadinstitute/hellbender/engine/filters/IntervalOverlapReadFilter.java]: I'm very surprised that this worked... but it certainly seems too.

**Pull request merged**